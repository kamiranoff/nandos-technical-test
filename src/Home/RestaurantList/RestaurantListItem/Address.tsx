import React, { FC } from 'react';
import { StyleSheet, Text, View } from 'react-native';

export interface IAddress {
  addressLocality: string;
  postalCode: string;
  streetAddress: string;
}

const Address: FC<IAddress> = ({
  addressLocality,
  postalCode,
  streetAddress,
}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{addressLocality}</Text>
      <Text style={styles.text}>{postalCode}</Text>
      <Text style={styles.text} numberOfLines={2}>
        {streetAddress}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    maxWidth: '75%',
  },
  text: {
    color: '#333',
  },
});

export default Address;
