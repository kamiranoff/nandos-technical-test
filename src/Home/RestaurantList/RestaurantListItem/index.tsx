import React, { FC } from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import Address from './Address';
import { RestaurantPayload } from '../../../types/Api';

export interface IRestaurantListItem {
  title: string;
  url: string;
  onPress: (url: string) => void;
  address: RestaurantPayload['geo']['address'];
}

const RestaurantListItem: FC<IRestaurantListItem> = ({
  title,
  url,
  onPress,
  address,
}) => (
  <View style={styles.container}>
    <View style={styles.subContainer}>
      <Text style={styles.title}>{title}</Text>
      <Pressable
        onPress={() => onPress(url)}
        android_ripple={{ color: '#d1011b', borderless: true }}>
        <Text style={styles.text}>Open Link</Text>
      </Pressable>
    </View>
    <Address
      addressLocality={address.addressLocality}
      postalCode={address.postalCode}
      streetAddress={address.streetAddress}
    />
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 10,
    paddingVertical: 20,
    borderBottomColor: '#CCC',
    borderBottomWidth: 0.5,
  },
  subContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  text: {
    color: '#0000EE',
  },
});

export default RestaurantListItem;
