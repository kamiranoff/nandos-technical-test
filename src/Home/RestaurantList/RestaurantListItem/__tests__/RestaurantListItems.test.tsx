import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import RestaurantListItem from '../index';

it('RestaurantListItem renders correctly', () => {
  const tree = renderer
    .create(
      <RestaurantListItem
        onPress={() => null}
        title="title"
        url="https://testurl"
        address={{
          addressLocality: 'London',
          streetAddress: 'westland place',
          postalCode: 'n17lp',
        }}
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('RestaurantListItem title and link text should be correct', () => {
  const wrapper = renderer.create(
    <RestaurantListItem
      onPress={() => null}
      title="My Restaurant"
      url="https://testurl"
      address={{
        addressLocality: 'London',
        streetAddress: 'westland place',
        postalCode: 'n17lp',
      }}
    />,
  );
  const texts = wrapper.root.findAllByType(Text);
  expect(texts[0].props.children).toBe('My Restaurant');
  expect(texts[1].props.children).toBe('Open Link');
});
