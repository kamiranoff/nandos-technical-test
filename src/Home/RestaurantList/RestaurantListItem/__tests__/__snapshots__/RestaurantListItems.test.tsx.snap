// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantListItem renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#DDD",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingVertical": 20,
    }
  }
>
  <Text>
    title
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#0000EE",
        }
      }
    >
      Open Link
    </Text>
  </View>
</View>
`;
